test/Microsoft.ML.Benchmarks/Program.cs

```diff
@@ -28,52 +30,33 @@ static void Main(string[] args)
         private static IConfig CreateCustomConfig() 
             => DefaultConfig.Instance
                 .With(Job.Default
+                    .WithWarmupCount(1) // for our time consuming benchmarks 1 warmup iteration is enough
                     .WithMaxIterationCount(20)
-                    .With(InProcessToolchain.Instance))
-                .With(new ClassificationMetricsColumn("AccuracyMacro", "Macro-average accuracy of the model"))
+                    .With(CreateToolchain()))
+                .With(new ExtraMetricColumn())
                 .With(MemoryDiagnoser.Default);
 
-        internal static string GetDataPath(string name)
-            => Path.GetFullPath(Path.Combine(_dataRoot, name));
-
-        static readonly string _dataRoot;
-        static Program()
+        /// <summary>
+        /// we need our own toolchain because MSBuild by default does not copy recursive native dependencies to the output
+        /// </summary>
+        private static IToolchain CreateToolchain()
         {
-            var currentAssemblyLocation = new FileInfo(typeof(Program).Assembly.Location);
-            var rootDir = currentAssemblyLocation.Directory.Parent.Parent.Parent.Parent.FullName;
-            _dataRoot = Path.Combine(rootDir, "test", "data");
+            var csProj = CsProjCoreToolchain.Current.Value;
+            var tfm = NetCoreAppSettings.Current.Value.TargetFrameworkMoniker;
+
+            return new Toolchain(
+                tfm, 
+                new ProjectGenerator(tfm), 
+                csProj.Builder, 
+                csProj.Executor);
         }
-    }
-
-    public class ClassificationMetricsColumn : IColumn
-    {
-        private readonly string _metricName;
-        private readonly string _legend;
 
-        public ClassificationMetricsColumn(string metricName, string legend)
+        internal static string GetInvariantCultureDataPath(string name)
         {
-            _metricName = metricName;
-            _legend = legend;
-        }
-
-        public string ColumnName => _metricName;
-        public string Id => _metricName;
-        public string Legend => _legend;
-        public bool IsNumeric => true;
-        public bool IsDefault(Summary summary, BenchmarkCase benchmark) => true;
-        public bool IsAvailable(Summary summary) => true;
-        public bool AlwaysShow => true;
-        public ColumnCategory Category => ColumnCategory.Custom;
-        public int PriorityInCategory => 1;
-        public UnitType UnitType => UnitType.Dimensionless;
+            // enforce Neutral Language as "en-us" because the input data files use dot as decimal separator (and it fails for cultures with ",")
+            Thread.CurrentThread.CurrentCulture = CultureInfo.InvariantCulture;
```

Comment:
This line of code is a bit surprising in a method that is supposed to return a data path.  Maybe it would be better to do this in the `Main` method, or a `GlobalSetup` method?

Reply:
@eerhardt I agree that I am breaking CQRS here. My only excuse is that I have named the method `GetInvariantCultureDataPath` so people can expect that.

I was thinking about moving it to a `[GlobalSetup]` method but I am afraid that people will don't follow this pattern in new benchmarks. By having it here I guarantee that whoever is going to use files will be using `CultureInfo.InvariantCulture` for reading these files.

I also wonder how ML.NET samples deal with the culture info problem. Does anybody know?